pkg/pool-stable/contracts/StablePool.sol

```diff
@@ -101,19 +117,16 @@ contract StablePool is IStablePool, BalancerPoolToken, BasePoolAuthentication, P
     error AmpUpdateNotStarted();
 
     /**
-     * @notice Parameters used to deploy a new Stable Pool.
-     * @param name ERC20 token name
-     * @param symbol ERC20 token symbol
-     * @param amplificationParameter Controls the "flatness" of the invariant curve. higher values = lower slippage,
-     * and assumes prices are near parity. lower values = closer to the constant product curve (e.g., more like a
-     * weighted pool). This has higher slippage, and accommodates greater price volatility
-     * @param version The stable pool version
+     * @notice Allow the swap manager to change the amplification parameter.
+     * @dev Unlike the swap fee percentage setting permission, this is non-exclusive.
      */
-    struct NewPoolParams {
-        string name;
-        string symbol;
-        uint256 amplificationParameter;
-        string version;
+    modifier authenticateByRole() {
+        // Allow if this is the swapFeeManager.
+        if (msg.sender != _vault.getPoolRoleAccounts(address(this)).swapFeeManager) {
+            // Otherwise, defer to governance.
+            _authenticateCaller();
+        }
+        _;
```

Comment:
This pattern is slightly different from the one we use for swap fees though.

In this case governance can always allow the permissioned function to be protected even if there's a designated swap fee manager. For the swap fee percentage in the vault admin, if the pool has a designated swap fee manager then governance is out of the loop.

I'm not against this pattern, but I'd double check whether it's what we want or not.

Reply:
Yes; I noted this in the docs, thinking it was being done by governance now, and they wanted the swap manager to also be able to do it as a convenience. We need to check with whoever wanted this change to make sure we have the correct requirements.